src/components/Pages/HomePage.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect, useRef, useCallback } from 'react'
+import useFetchImages from '../../hooks/useFetchImages'
+import ScrollToTopButton from '../ScrollToTopButton'
+import Title from '../Title'
+import Loading from '../Loading'
+import Card from '../Card'
+
+function HomePage() {
+  const [likedImages, setLikedImages] = useState<ImageTypes[]>(
+    JSON.parse(localStorage.getItem('nasa-liked-images') || '[]')
+  )
+  const [needMoreImages, setNeedMoreImages] = useState(true)
+  const { images, isLoading } = useFetchImages(needMoreImages)
+
+  // Observing Last Element on Page to create infinite scroll
+  const observer: React.MutableRefObject<any> = useRef()
+  const lastImageElementRef = useCallback(
+    (element: HTMLDivElement) => {
+      if (observer.current) observer.current.disconnect()
+      observer.current = new IntersectionObserver((entries) => {
+        if (entries[0].isIntersecting) setNeedMoreImages(true)
```

Comment:
Also want to prevent triggering a fetch again when already in a loading state.
```suggestion
        if (!isLoading && entries[0].isIntersecting) setNeedMoreImages(true)
```